m4/ac_define_dir.m4

```diff
@@ -0,0 +1,34 @@
+dnl @synopsis AC_DEFINE_DIR(VARNAME, DIR [, DESCRIPTION])
```

Comment:
This whole m4 macro is uneeded by linking the Lua sources into `dyz` as suggested.